### PR TITLE
Feature : generation docker-compose by maven

### PR DESCRIPTION
**Pull request by Silicoman**
When a user executes "mvn install" , he generates a docker-compose at  folder target/extra-resources/  (parent level && android module) 
**With dynamic variables :**
- version server sonarqube describes inside pom parent
- versions ecocode plugins describes by  pom parent
- absolute path to target folder plugin

Due to lack gradle integration with mvn. **NEED validation of android-plugin** .

_Keep original docker-compose.yml until end hackathon to not lost people. Doc will be change at this time._

### CODE REVIEW

File: src/android-plugin/src/main/scripts/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '3'
```
Comment (MP-Aubay) [inline src/android-plugin/src/main/scripts/docker-compose.yml:1]: Maybe this file should be deleted ? I do not think it is usefull to have a specific docker-compose file by plugin

**Review by MP-Aubay — COMMENTED** [review]
To avoid confusion it could be necessary to update documentation in this PR and remove the existing docker-compose.yml file.

The next step could be to launch docker-compose command from maven, but it could be done in another PR.

**Comment by dedece35** [comment]
hello @Silicoman, @MP-Aubay,

it's a good idea to improve docker installation system. But there is another PR with the same kind of idea : see #89 
Maybe merge these two ideas into another new PR in order to improve our docker installation system.
(please see PR #89 created by @obeone)

My feedbacks of this PR : 
- I agree with @MP-Aubay, for me, it is'nt a optimum solution to have a custom docker-compose file by plugin but it would better to have only one to do all.
- good idea to manage sonarqube version on pom.xml files

**Comment by obeone** [comment]
Hi,

I choose a different approch for PR #137 : 
- It use a Maven docker official image to build every plugins
- After that, it use a clean (with build files) official SonarQube and add ecoCode's jar file in it
- Having a GitHub Workflow to automatically build on every new version

The idea for me is having ecoCode repository offering an already built image to allow a quick start (and especially a quick test, without the need to install JDK etc...). 
And more than that, having just one common build usable by everyone... It's more ecoCode, no ? ;)

The PR include modifications to docker-compose.yml to use the prebuilt container

If you want to test it : 

```shell
docker run -d --pull always \
       -v sq_ecocode_extensions:/opt/sonarqube/extensions \
       -v sq_ecocode_logs:/opt/sonarqube/logs \
       -v sq_ecocode_data:/opt/sonarqube/data \
       -p 9000:9000 \
       --name sonarqube-ecocode \
       ghcr.io/obeone/ecocode:latest
```

**Comment by dedece35** [comment]
Hi, my answer in the other PR #137

**Comment by dedece35** [comment]
Hi @Silicoman ,
we are migrating this repo to a new one : please see https://github.com/green-code-initiative/ecocode/pull/9
that's why I change this PR to "draft".

**Pull request closed without merging**